svir/dialogs/drive_oq_engine_server_dialog.py

```diff
@@ -140,7 +140,11 @@ def __init__(self, iface, viewer_dock):
         # keep track of the log lines acquired for each calculation
         self.calc_log_line = {}
         self.session = None
-        self.hostname = None
+        self.hostname = hostname
+        if hostname is not None:
+            self.forced_hostname = True
+        else:
```

Comment:
Why not just use the hostname in the `ifs` ? The flag forced_hostname does not do much,

Reply:
It's a bit tricky. While playing with the plugin, the user might use the plugin settings dialog to set up the connection to a different engine server. If it happens, then when the dialog that drives the engine attempts to login to the engine server, we must get the modified user credentials (see below).
I am adding the hostname parameter here to be able to call this from tests, forcing the hostname to remain as specified, instead of reading it from the user settings.